distributed/scheduler.py

```diff
@@ -2317,18 +2317,19 @@ def transition_processing_erred(
 
             if self.validate:
                 assert cause or ts.exception_blame
-                assert ts.processing_on
                 assert not ts.who_has
                 assert not ts.waiting_on
 
+            maybe_stale_ws = ts.processing_on
+            assert maybe_stale_ws
+
             if ts.actor:
-                assert ts.processing_on
-                ws = ts.processing_on
-                ws.actors.remove(ts)
+                maybe_stale_ws.actors.remove(ts)
 
-            w = _remove_from_processing(self, ts)
+            _remove_from_processing(self, ts)
 
-            ts.erred_on.add(w)
+            # NOTE: `worker` is None if it died, i.e. `KilledWorker`
+            ts.erred_on.add(worker or maybe_stale_ws.address)
```

Comment:
Having two possible sources for the worker address we use here still feels like a consistency issue. Why not just use `maybe_stale_ws.address`?

Reply:
Because at the moment, it's possible that the task ran (and failed) on a worker we didn't expect it to. Since this PR was making the `erred` case consistent with `memory`, it's possible `worker != maybe_stale_ws.address`, in which case `worker` is the correct address since that's where the task actually ran and failed.